### PR TITLE
Feature/create data source

### PR DESCRIPTION
**Pull request by andreyPromaster**
- Create data-driver to get story data from **DynamoDB**
- Create schemas

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,38 +3,31 @@ repos:
     rev: v1.2.3
     hooks:
       - id: trailing-whitespace
-        files: ^(packages|tests)/
         types: [file, python]
```
Comment (vadim-denisov) [inline .pre-commit-config.yaml:6]: I suggest that trailing-whitespace be removed in all files, not just python

File: app.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import boto3
```
Comment (vadim-denisov) [inline app.py:1]: I recommend to store code in story_game/...

File: app.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from conf import settings
+from entities.connection import DynamoDBDriver
+
+
+def test():
+    conf = Config(retries={"max_attempts": 1, "mode": "standard"})
+
+    conn = boto3.resource(
+        "dynamodb",
+        endpoint_url=f"http://{settings.DYNAMODB_HOST}:{settings.DYNAMODB_PORT}",
```
Comment (vadim-denisov) [inline app.py:13]: `endpoint_url` can be property of a setting object

File: app.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+from botocore.config import Config
+
+from conf import settings
+from entities.connection import DynamoDBDriver
+
+
+def test():
+    conf = Config(retries={"max_attempts": 1, "mode": "standard"})
+
+    conn = boto3.resource(
```
Comment (vadim-denisov) [inline app.py:11]: And we should store connection in module scope in order not to create connection every time when we call a lambda function

File: entities/connection.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+from boto3.dynamodb.conditions import Key
+
+from conf import settings
+from entities.schemas import Node, Story
+
+
+class DataDriver(abc.ABC):
+    @abc.abstractmethod
+    def get_story(self, story_id: str):
+        pass
```
Comment (vadim-denisov) [inline entities/connection.py:13]: Use docstring with description of this method instead of `pass`

File: pyproject.toml [hunk 6]
```diff
@@ -6,6 +6,9 @@ authors = ["andreyPromaster <and45.rew@ya.ru>"]
 
 [tool.poetry.dependencies]
 python = "^3.8"
+Flask = "^2.0.2"
+pydantic = {extras = ["email", "dotenv"], version = "^1.8.2"}
```
Comment (vadim-denisov) [inline pyproject.toml:10]: Why extras?

Comment (andreyPromaster) [reply pyproject.toml:10]: `pydantic[dotenv]` provides optional  dotenv file support with Settings. `email `is redundant. i will remove it/

File: .pre-commit-config.yaml (hunk 1 above)
Comment (vadim-denisov) [reply .pre-commit-config.yaml]: @andreyPromaster What about other `types: [file, python]`?

File: entities/connection.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            FilterExpression=f"attribute_exists(nodes.{uri})",
+            ProjectionExpression=f"nodes.{uri}",
+        )
+        if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+            raise DynamoDBError(json.dumps(response))
+
+        if response["Count"] == 1:
+            return Node(**response["Items"][0]["nodes"][uri])
+        else:
+            return {}
```
Comment (vadim-denisov) [inline entities/connection.py:48]: I think returning `None` is more natural

File: entities/connection.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    def get_story(self, story_id: str):
+        response = self.connection.query(
+            KeyConditionExpression=Key("id").eq(story_id),
+            ProjectionExpression="id, root",
+        )
+        if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+            raise Exception(json.dumps(response))
+
+        if response["Count"] == 1:
+            return Story(**response["Items"][0])
```
[the remark this answers is not in the document]
Comment (andreyPromaster) [reply entities/connection.py:34]: > Is it case when we have HTTP Status 200 and Count 0?
When count is 0 response will be 404

File: .pre-commit-config.yaml (hunk 1 above)
Comment (andreyPromaster) [reply .pre-commit-config.yaml]: I removed `types` from this hook

**Review by vadim-denisov — APPROVED** [review]

**Pull request merged**